### PR TITLE
{HDInsight} new api version 2024-08-01-preview

### PR DESCRIPTION
**Pull request by yuc-Li**
**Related command**
`az hdinsight create`
`az hdinsight update`
`az hdinsight azure-monitor-agent show`
`az hdinsight azure-monitor-agent enable`
`az hdinsight azure-monitor-agent disable`

**Description**
Update api version to 2024-08-01-preview. 
Added params for create cluster. 
Added params for update cluster.
Added 3 command for manage Azure Monitor Agent on the cluster.

**Testing Guide**
Create cluster with new param:
- az hdinsight create --public-ip-tag-type FirstPartyUsage --public-ip-tag-value HDInsight 'other params'

Update cluster identity:
- az hdinsight update -g group -n clustername --assign-identity /subscriptions/000000000000000000000000000000/resourceGroups/ps-test/providers/Microsoft.ManagedIdentity/userAssignedIdentities/hdi-msi

Manage Azure Monitor Agent:
- az hdinsight azure-monitor-agent enable -n {cluster} --resource-group {rg} --workspace {workspace_id} --primary-key {workspace_key}
- az hdinsight azure-monitor-agent show -n {cluster} --resource-group {rg}
- az hdinsight azure-monitor-agent disable -n {cluster} --resource-group {rg}

**History Notes**

[HDInsight] `az hdinsight create`: Support setting IP tags when creating HDInsight cluster
[HDInsight] `az hdinsight update`: Support update the managed identity of the cluster
[HDInsight] `az hdinsight azure-monitor-agent show`: Get update the managed identity of the cluster
[HDInsight] `az hdinsight azure-monitor-agent enable`: Enable the Azure Monitor Agent logs integration on an HDInsight cluster
[HDInsight] `az hdinsight azure-monitor-agent disable`: Disable the Azure Monitor Agent logs integration on an HDInsight cluster

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
HDInsight

**Comment by yuc-Li** [comment]
Hi @kairu-ms , could you help review PR and there have a error in pipeline, I checked 'get_azure_monitor_agent_status' existed in ExtensionsOperations. 
image

> [notice] A new release of pip is available: 23.0.1 -> 24.2
[notice] To update, run: pip install --upgrade pip
WARNING: This command has been deprecated and will be removed in a future release.
ERROR: Error occurred loading commands!
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: The operation 'azure.mgmt.hdinsight.operations#ExtensionsOperations.get_azure_monitor_agent_status' is invalid.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/azure/cli/core/commands/command_operation.py", line 61, in get_op_handler
    handler = getattr(handler, part)
AttributeError: type object 'ExtensionsOperations' has no attribute 'get_azure_monitor_agent_status'

**Comment by yuc-Li** [comment]
Hi @jsntcy @kairu-ms , request a review.

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/setup.py [hunk 1]
```diff
@@ -86,7 +86,7 @@
     'azure-mgmt-eventgrid==10.2.0b2',
     'azure-mgmt-eventhub~=10.1.0',
     'azure-mgmt-extendedlocation==1.0.0b2',
-    'azure-mgmt-hdinsight~=9.0.0',
```
Comment (kairu-ms) [inline src/azure-cli/setup.py:89]: the next beta version should be 9.0.1b1 as you already released the 9.0.0 stable version

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**